### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mish-sk**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,44 @@
...
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float,
+                 clean_power,
+                 average_rating: float,
+                 count_of_ratings):
```
Comment (Abnormaltype) [inline app/main.py:12]: add type annotation

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,44 @@
...
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list):
```
Comment (Abnormaltype) [inline app/main.py:18]: add type annotation

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,44 @@
...
+        return sum(income)
+
+    def calculate_washing_price(self, car):
+        price = (car.comfort_class
+                 * (self.clean_power - car.clean_mark)
+                 * self.average_rating
+                 / self.distance_from_city_center)
+        return round(price, 1)
+
+    def wash_single_car(self, car):
```
Comment (Abnormaltype) [inline app/main.py:31]: add type annotation

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,44 @@
...
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list):
+        income = []
+        for car in cars:
+            income.append(self.wash_single_car(car))
+        return sum(income)
```
Comment (Abnormaltype) [inline app/main.py:22]: in this method, you should call `wash_single_car` method

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,44 @@
...
+        return round(price, 1)
+
+    def wash_single_car(self, car):
+        single_car_price = 0
+        if car.clean_mark < self.clean_power:
+            single_car_price = self.calculate_washing_price(car)
+            car.clean_mark = self.clean_power
+        return single_car_price
+
+    def rate_service(self, rate):
```
Comment (Abnormaltype) [inline app/main.py:38]: add type annotation

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 4 above)
Comment (mish-sk) [reply app/main.py:22]: why is this doesn't count as calling `wash_single_car` method? It is in the method

Comment (Abnormaltype) [reply app/main.py:22]: hm, okay, you just replace logic, got it. Fix type annotation

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 6]
```diff
@@ -1,8 +1,44 @@
...
+        return round(price, 1)
+
+    def wash_single_car(self, car: list) -> float:
+        single_car_price = 0
+        if car.clean_mark < self.clean_power:
+            single_car_price = self.calculate_washing_price(car)
+            car.clean_mark = self.clean_power
+        return single_car_price
+
+    def rate_service(self, rate: int):
```
Comment (Abnormaltype) [inline app/main.py:38]: add return type

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 7]
```diff
@@ -1,8 +1,44 @@
...
+        return round(price, 1)
+
+    def wash_single_car(self, car: list) -> float:
+        single_car_price = 0
+        if car.clean_mark < self.clean_power:
+            single_car_price = self.calculate_washing_price(car)
+            car.clean_mark = self.clean_power
+        return single_car_price
+
+    def rate_service(self, rate: int) -> int:
```
Comment (Abnormaltype) [inline app/main.py:38]:
```suggestion
    def rate_service(self, rate: int) -> None:
```

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Review by Abnormaltype — APPROVED** [review]
Well done!

**Pull request closed without merging**